components/two-factor-authentication/TwoFactorAuthenticationModal.tsx

```diff
@@ -0,0 +1,307 @@
+                defaultMessage="To enable Two-Factor Authentication (2FA), follow the steps <link>here</link>"
+                values={{
+                  link: getI18nLink({
+                    href: getSettingsRoute(LoggedInUser.collective, 'user-security'),
+                    as: Link,
+                  }),
+                }}
+              />
+            </P>
+          </Flex>
+        )}
+
+        {selectedMethod === 'recovery_code' && (
+          <RecoveryCodeOptions value={twoFactorCode} onChange={setTwoFactorCode} disabled={confirming} />
+        )}
+
+        {(selectedMethod === 'yubikey_otp' || selectedMethod === 'totp') && (
+          <AuthenticatorOption
+            value={twoFactorCode}
+            onChange={setTwoFactorCode}
+            supportedMethods={supportedMethods}
+            disabled={confirming}
+          />
+        )}
+
+        {selectedMethod === 'webauthn' && <WebauthnOption />}
+      </Box>
+
+      {supportedMethods.length > 1 && (
+        <Box mt={4}>
+          <FormattedMessage defaultMessage="You can also use alternative methods:" />
+          <ul>
+            {alternativeMethods.includes('totp') && (
+              <li>
+                <StyledLinkButton onClick={() => setSelectedMethod('totp')}>
+                  <FormattedMessage defaultMessage="Authenticator Code" />
+                </StyledLinkButton>
+              </li>
+            )}
+            {alternativeMethods.includes('webauthn') && (
+              <li>
+                <StyledLinkButton onClick={() => setSelectedMethod('webauthn')}>
+                  <FormattedMessage defaultMessage="U2F (Hardware Key, Passkey, Phone, etc)" />
+                </StyledLinkButton>
+              </li>
+            )}
+            {alternativeMethods.includes('recovery_code') && (
+              <li>
+                <StyledLinkButton onClick={() => setSelectedMethod('recovery_code')}>
+                  <FormattedMessage defaultMessage="Recovery code" />
+                </StyledLinkButton>
+              </li>
+            )}
+          </ul>
+        </Box>
+      )}
+
+      <ModalFooter isFullWidth dividerMargin="1rem 0">
+        <Flex justifyContent="right" flexWrap="wrap">
+          {cancellable && (
+            <StyledButton disabled={confirming} mr={2} minWidth={120} onClick={cancel}>
+              <FormattedMessage id="actions.cancel" defaultMessage="Cancel" />
+            </StyledButton>
+          )}
+          <StyledButton
+            ml={2}
+            minWidth={120}
+            buttonStyle="primary"
+            loading={confirming}
+            disabled={!verifyBtnEnabled}
+            onClick={selectedMethod === 'webauthn' ? useWebAuthn : confirm}
+          >
+            {selectedMethod === 'recovery_code' ? (
+              <FormattedMessage id="login.twoFactorAuth.reset" defaultMessage="Reset 2FA" />
+            ) : (
+              <FormattedMessage id="actions.verify" defaultMessage="Verify" />
```

Comment:
I feel like we could straight call `useWebAuthn` if there's a method set up, clicking verify is a bit confusing considering we're offering other methods.